service/controller/resource/certificates/resource.go

```diff
@@ -117,10 +139,15 @@ func (r *Resource) getSource(ctx context.Context, v interface{}) (*corev1.Secret
 
 		secret, err = r.k8sClient.K8sClient().CoreV1().Secrets(secretNamespace).Get(ctx, secretName, metav1.GetOptions{})
 		if apierrors.IsNotFound(err) {
-			// fall through
+			secret = nil
 		} else if err != nil {
 			return nil, microerror.Mask(err)
 		}
+
+		if secret != nil {
+			// We return the first secret we find
```

Comment:
What's the idea here?

Reply:
We have multiple sources for secrets now, the one in the default namespace, or the one for CAPI, once we find one, we return to avoid looping for no reason